libdrgn/python/thread.c

```diff
@@ -0,0 +1,114 @@
+#include "drgnpy.h"
+
+static Program *Thread_prog(Thread *self)
+{
+	return container_of(self->thread.prog, Program, prog);
+}
+
+PyObject *Thread_wrap(struct drgn_thread *thread)
+{
+	Thread *ret = (Thread *)Thread_type.tp_alloc(&Thread_type, 0);
+	if (!ret)
+		return NULL;
+	struct drgn_error *err =
+		drgn_thread_dup_internal(thread, &ret->thread);
+	if (err) {
+		ret->thread.prog = NULL;
+		Py_DECREF(ret);
+		return set_drgn_error(err);
+	}
+	Py_INCREF(container_of(thread->prog, Program, prog));
+	return (PyObject *)ret;
+}
+
+static void Thread_dealloc(Thread *self)
+{
+	if (self->thread.prog) {
+		Program *prog = Thread_prog(self);
+		drgn_thread_deinit(&self->thread);
+		Py_DECREF(prog);
+	}
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static PyObject *Thread_get_tid(Thread *self)
+{
+	return PyLong_FromUnsignedLong(self->thread.tid);
+}
+
+static DrgnObject *Thread_get_object(Thread *self)
+{
+	const struct drgn_object *object;
+	struct drgn_error *err = drgn_thread_object(&self->thread, &object);
+	if (err)
+		return set_drgn_error(err);
+	DrgnObject *ret = DrgnObject_alloc(Thread_prog(self));
```

Comment:
Missing `NULL` check for `ret`
```suggestion
	DrgnObject *ret = DrgnObject_alloc(Thread_prog(self));
	if (!ret)
	    return NULL;
	
```